core/src/apresolve.rs

```diff
@@ -82,30 +101,37 @@ impl ApResolver {
         let result = self.try_apresolve().await;
 
         self.lock(|inner| {
-            let data = match result {
-                Ok(data) => data,
-                Err(e) => {
-                    warn!("Failed to resolve access points, using fallbacks: {}", e);
-                    ApResolveData::default()
-                }
+            let (data, error) = match result {
+                Ok(data) => (data, None),
+                Err(e) => (ApResolveData::default(), Some(e)),
             };
 
-            inner.data.accesspoint = self.process_data(data.accesspoint);
-            inner.data.dealer = self.process_data(data.dealer);
-            inner.data.spclient = self.process_data(data.spclient);
+            inner.data.accesspoint = self.process_ap_strings(data.accesspoint);
+            inner.data.dealer = self.process_ap_strings(data.dealer);
+            inner.data.spclient = self.process_ap_strings(data.spclient);
+
+            if inner.data.is_any_empty() {
+                if let Some(error) = error {
+                    warn!(
+                        "Failed to resolve all access points, using fallbacks: {}",
+                        error
+                    );
+                } else {
+                    warn!("Failed to resolve all access points, using fallbacks");
+                }
+                inner
+                    .data
+                    .extend_filter_port(AccessPoints::fallback(), self.port_config());
```

Comment:
Thinking about this. This method *only* for filtering the fallbacks, when the same logic to filter on ports is also in `process_ap_strings`? I'm thinking that should then also use the filter, or we should inject the fallbacks before the call to `process_ap_strings` like it was before.

Reply:
Good point. I think I went a bit off track here, I'll fix this with the next commit

Reply:
👍  keep it coming